### PR TITLE
Add logging in HttpManager plus minor logging updates in RegionDiscoveryProvider.

### PR DESCRIPTION
**Pull request by pmaytak**
Add logging in HttpManager plus minor logging updates in RegionDiscoveryProvider.

### CODE REVIEW

File: src/client/Microsoft.Identity.Client/Region/RegionDiscoveryProvider.cs [hunk 1]
```diff
@@ -86,7 +86,7 @@ private async Task<string> GetRegionAsync(RequestContext requestContext)
 
             if (!string.IsNullOrEmpty(region))
             {
-                requestContext.Logger.Info($"[Region discovery] Region found in environment variable: {region}.");
+                requestContext.Logger.Info($"[Region Discovery] Region found in environment variable: {region}.");
```
Comment (jmprieur) [inline src/client/Microsoft.Identity.Client/Region/RegionDiscoveryProvider.cs:89]: why a capitalized D? (this is a sentence?). Microsoft is advising to use sentence casing in public facing docs, logs etc ...

**Review by jmprieur — COMMENTED** [review]
LGTM for the wording.
I think we should use sentence casing (according to [Microsoft voice](https://docs.microsoft.com/en-us/style-guide/top-10-tips-style-voice))

File: src/client/Microsoft.Identity.Client/Region/RegionDiscoveryProvider.cs (hunk 1 above)
Comment (pmaytak) [reply src/client/Microsoft.Identity.Client/Region/RegionDiscoveryProvider.cs:89]: Capitalized for consistency. We used `[Region Discovery]` and `[Instance Discovery]` else where. I wasn't aware of that guidance, so I can update to use sentence casing.

Comment (bgavrilMS) [reply src/client/Microsoft.Identity.Client/Region/RegionDiscoveryProvider.cs:89]: @jmprieur - these are MSAL logs, not docs.

File: src/client/Microsoft.Identity.Client/Http/HttpManager.cs [hunk 2]
```diff
@@ -211,6 +212,8 @@ private static bool HasRetryAfterHeader(HttpResponse response)
                 requestMessage.Method = method;
                 requestMessage.Content = body;
 
+                logger.Verbose($"Sending request from HttpManager. Method: {method}. URI path: {endpoint.Scheme}://{endpoint.Authority}{endpoint.AbsolutePath}. ");
```
Comment (bgavrilMS) [inline src/client/Microsoft.Identity.Client/Http/HttpManager.cs:215]: Do you want to log the HTTP status of the response as well?

**Review by bgavrilMS — APPROVED** [review]

**Review by neha-bhargava — APPROVED** [review]

File: src/client/Microsoft.Identity.Client/Http/HttpManager.cs (hunk 2 above)
Comment (pmaytak) [reply src/client/Microsoft.Identity.Client/Http/HttpManager.cs]: Done. We also have additional logging for response errors in the outer method.

**Pull request merged**